state/podspec_ops.go

```diff
@@ -73,9 +73,9 @@ func (op *setPodSpecOperation) buildTxn(_ int) ([]txn.Op, error) {
 	if err != nil {
 		return nil, errors.Annotate(err, "setting pod spec")
 	}
-	if app.Life() != Alive {
+	if app.Life() == Dead {
 		return nil, errors.Annotate(
-			errors.Errorf("application %s not alive", appName),
+			errors.NotValidf("application %s is dead", appName),
```

Comment:
this will result in a malformed error message
instead something like:
errors.NotValidf("setting pod-spec on dead application %s", appName)